packages/cashier/src/constants/transaction-status.tsx

```diff
@@ -94,5 +95,16 @@ export const getStatus = (transaction_hash, transaction_type, status_code) => {
         },
     };
 
-    return status_list[transaction_type][formatted_status_code];
+    const isDeposit = (status: TStatusCode): status is keyof typeof status_list.deposit =>
+        Object.keys(status_list.deposit).includes(status);
+    const isWithdrawal = (status: TStatusCode): status is keyof typeof status_list.withdrawal =>
+        Object.keys(status_list.withdrawal).includes(status);
+
+    if (transaction_type === 'deposit' && isDeposit(status_code)) {
+        return status_list[transaction_type][status_code];
+    } else if (transaction_type === 'withdrawal' && isWithdrawal(status_code)) {
+        return status_list[transaction_type][status_code];
+    }
+
+    return null;
```

Comment:
This here has some implications
Basically, the return value of `getStatus` is always possibly `null`, which makes it so that we have TS errors like the ones on the screenshot below.


<img width="979" alt="Screenshot 2023-04-18 at 14 47 50" src="https://user-images.githubusercontent.com/119863957/232767945-a3af68f8-8b3b-408c-ac7a-07c3b522bee0.png">